src/EnergyPlus/HVACUnitaryBypassVAV.cc

```diff
@@ -3542,61 +3434,68 @@ namespace HVACUnitaryBypassVAV {
                     }
                 } else if (CBVAV(CBVAVNum).NumZonesCooled > CBVAV(CBVAVNum).NumZonesHeated) {
                     if (QZoneReqCool < 0.0) {
-                        QZoneReq = QZoneReqCool;
                         CBVAV(CBVAVNum).HeatCoolMode = CoolingMode;
                     } else if (QZoneReqHeat > 0.0) {
-                        QZoneReq = QZoneReqHeat;
                         CBVAV(CBVAVNum).HeatCoolMode = HeatingMode;
-                    } else {
-                        QZoneReq = 0.0;
                     }
                 } else {
                     if (std::abs(QZoneReqCool) > std::abs(QZoneReqHeat) && QZoneReqCool != 0.0) {
-                        QZoneReq = QZoneReqCool;
                         CBVAV(CBVAVNum).HeatCoolMode = CoolingMode;
                     } else if (std::abs(QZoneReqCool) < std::abs(QZoneReqHeat) && QZoneReqHeat != 0.0) {
-                        QZoneReq = QZoneReqHeat;
                         CBVAV(CBVAVNum).HeatCoolMode = HeatingMode;
                     } else if (std::abs(QZoneReqCool) == std::abs(QZoneReqHeat) && QZoneReqCool != 0.0) {
-                        QZoneReq = QZoneReqCool;
                         CBVAV(CBVAVNum).HeatCoolMode = CoolingMode;
-                    } else {
-                        QZoneReq = 0.0;
+                    }
+                }
+            } else if (SELECT_CASE_var == LoadPriority) {
+                if (std::abs(QZoneReqCool) > std::abs(QZoneReqHeat) && QZoneReqCool != 0.0) {
+                    CBVAV(CBVAVNum).HeatCoolMode = CoolingMode;
+                } else if (std::abs(QZoneReqCool) < std::abs(QZoneReqHeat) && QZoneReqHeat != 0.0) {
+                    CBVAV(CBVAVNum).HeatCoolMode = HeatingMode;
+                } else if (CBVAV(CBVAVNum).NumZonesHeated > CBVAV(CBVAVNum).NumZonesCooled) {
+                    if (QZoneReqHeat > 0.0) {
+                        CBVAV(CBVAVNum).HeatCoolMode = HeatingMode;
+                    } else if (QZoneReqCool < 0.0) {
+                        CBVAV(CBVAVNum).HeatCoolMode = CoolingMode;
+                    }
+                } else if (CBVAV(CBVAVNum).NumZonesHeated < CBVAV(CBVAVNum).NumZonesCooled) {
+                    if (QZoneReqCool < 0.0) {
+                        CBVAV(CBVAVNum).HeatCoolMode = CoolingMode;
+                    } else if (QZoneReqHeat > 0.0) {
+                        CBVAV(CBVAVNum).HeatCoolMode = HeatingMode;
+                    }
+                } else {
+                    if (QZoneReqCool < 0.0) {
+                        CBVAV(CBVAVNum).HeatCoolMode = CoolingMode;
+                    } else if (QZoneReqHeat > 0.0) {
+                        CBVAV(CBVAVNum).HeatCoolMode = HeatingMode;
                     }
                 }
             }
         }
+
+        if (CBVAV(CBVAVNum).LastMode != CBVAV(CBVAVNum).HeatCoolMode) {
+            CBVAV(CBVAVNum).changeOverTimer = thisTime + CBVAV(CBVAVNum).minModeChangeTime;
+        }
```

Comment:
Timer is update to be the time of last mode change plus the minimum time in a given mode.